### ISSUE TITLE
Init attribute for options does not work

### ISSUE THREAD

**Issue by EdwardOst** [ko1nksm/getoptions#51]
My understanding (possibly incorrect) based on the Reference document is that options (options with an optional argument) should be able to be:

* set explicitly to some value passed on the actual command line (consistent with validation)
* set to "on" value if the option is present in the commandline and no value is passed, -e.g. -n
* set to the "no" value if the negated option is present in the commandline and no value is passed, e.g. +n
* if the option is not passed on the commandline my expectation was that the init attribute would be the default

It seems that the last bullet item is not correct.  The result is that options with optional values cannot have a default for when they are not present.  I have tried setting a default as an explicit string

    init:="missing value"

And I have tried setting it to one of the pre-defined expressions

    init:@on

Note that the latter is still limiting because it would limit the default value to be either the on or the no value rather than a distinct value for the case when the option was not specified.

I believe the init option is not working for options (with optional values) currently.  (It works fine for parameters)

Thanks again for this great tool.  As you can see from the example, I prefer to use it with functions in library mode.  In library mode I usually have strict mode enabled with "set -euo pipefail" enabled.  But I have tested with strict mode disabled and the result is the same.

[example.zip](https://github.com/ko1nksm/getoptions/files/11781497/example.zip)

**Comment by EdwardOst** [comment]
I spoke too soon.  It looks like init is not working for parameters either.  See other issue.

**Comment by EdwardOst** [comment]
I reviewed the generated code and I can see that there are some potential issues with trying use getoptions for functions.  Nothing insurmountable, but something for another issue.  So I re-tested with regular getoptions for commands.  I used the basics.sh from the examples section and modified it to test the same issues wrt init for both params and options.  It appears init does not work for either params or options.  Modified basic example and test code included below.

[basic-test.zip](https://github.com/ko1nksm/getoptions/files/11781965/basic-test.zip)
